src/main/scala/strawman/collection/mutable/ListBuffer.scala

```diff
@@ -12,7 +12,7 @@ import scala.Predef.{assert, intWrapper}
 
 /** Concrete collection type: ListBuffer */
 class ListBuffer[A]
-  extends Seq[A]
+  extends GrowableSeq[A]
```

Comment:
> Why do we need to separately say this is growable, buildable, and a builder?

Indeed we could probably assume that `Growable` imply `Buildable`. I see no example where we would have the former without also having the latter.

I decided to put the method that returns the builder in the companion object because I think that could be useful for other purposes (see #97).

> can't every mutable collection be treated as its own builder?

That’s a good question. Actually, @odersky initially designed mutable collections to be their own builders. I thought that it would make the hierarchy slightly more complicated (that’s yet another type that shows up in the linearization) and also I think the public `result` method of builders makes little sense on a concrete collection.